src/index_schema.cc

```diff
@@ -146,7 +146,7 @@ absl::StatusOr<std::shared_ptr<indexes::IndexBase>> IndexFactory(
 
 absl::StatusOr<std::shared_ptr<IndexSchema>> IndexSchema::Create(
     ValkeyModuleCtx *ctx, const data_model::IndexSchema &index_schema_proto,
-    vmsdk::ThreadPool *mutations_thread_pool, bool skip_attributes) {
+    vmsdk::ThreadPool *mutations_thread_pool, bool skip_attributes, bool reload) {
```

Comment:
Would it make more sense to just zero out the stats explicitly here before? 

https://github.com/valkey-io/valkey-search/blob/8b53ed55a179ffb076f30cdfd03a4e63ef859eb5/src/schema_manager.cc#L432

The parameter `reload` isn't very intuitive to me - I guess it is to differentiate between new index schemas and index schemas from RDB?